File: ska_frontend/src/modules/auth/dto_user_details.rs
use std::collections::HashSet;

use serde::{Deserialize, Serialize};
use strum_macros::{AsRefStr, Display, EnumString, IntoStaticStr};

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct DtoUserDetails {
    pub user_id: i64,
    pub sub: String,
    pub username: String,
    pub email: String,
    pub roles: HashSet<AuthRoles>,
}

#[derive(
    Clone,
    Debug,
    Serialize,
    Deserialize,
    AsRefStr,
    IntoStaticStr,
    EnumString,
    Display,
    PartialEq,
    Eq,
    Hash,
)]
#[strum(serialize_all = "SCREAMING_SNAKE_CASE")]
pub enum AuthRoles {
    #[serde(rename = "SKA_ADMIN")]
    SkaAdmin,
    #[serde(rename = "SKA_USER")]
    SkaUser,
    #[serde(rename = "SKA_GUEST")]
    SkaGuest,
}

File: ska_frontend2/src/domain/auth/dto_user_details.ts
export class DtoUserDetails {
    user_id: number;
    sub: string;
    username: string;
    email: string;
    roles: AuthRoles[];

    constructor(obj: {
        user_id: number,
        sub: string,
        username: string,
        email: string,
        roles: AuthRoles[],
    }) {
        this.user_id = obj.user_id;
        this.sub = obj.sub;
        this.username = obj.username;
        this.email = obj.email;
        this.roles = obj.roles;
    }
}

export enum AuthRoles {
    SKA_ADMIN = 'SKA_ADMIN',
    SKA_USER = 'SKA_USER',
    SKA_GUEST = 'SKA_GUEST',
}
